eclass/linux-info.eclass

```diff
@@ -696,11 +696,17 @@ linux-info_get_any_version() {
 		die "${FUNCNAME}() called on non-Linux system, please fix the ebuild"
 	fi
 
-	if ! get_version; then
+	if [[ ${MERGE_TYPE} == binary && -z ${LINUX_INFO_BINARY_RESET} ]]; then
+		unset KV_FULL _LINUX_CONFIG_EXISTS_DONE KV_OUT_DIR
+		LINUX_INFO_BINARY_RESET=1
```

Comment:
This LINUX_INFO_BINARY_RESET variable will persist in environment.bz2, and if someone uses quickpkg to generate a new binary package from it then the new binary package will inherit the LINUX_INFO_BINARY_RESET setting.

Reply:
Any suggestion for how to work around that?

I don't really care about quickpkg too much, but if there's an easy solution I am happy to make an adjustment.

Reply:
I will say that this scenario seems a bit unlikely: if the user already installed from a binpkg once, why would they use quickpkg to create a new binpkg?

Reply:
The quickpkg invocation could be against a package installed during a stage build or something like that where the quickpkg caller might not have access to the binary package anymore.

Anyway I agree quickpkg doesn't necessarily need to be supported.

I suppose you might trigger this code based on the EBUILD_PHASE variable, and then I realized that the environment is not actually saved for pkg_pretend, so you might want to run the code again in pkg_setup. I also wonder about pkg_config.

Reply:
I always see quickpkg as a bit of hack that's more for emergencies/testing than using regularly with everything, lucky when it works but there's a lot of things that can break them in subtle ways (I know nvidia-drivers quickpkg been broken for ages and it would need a ebuild workaround, but I haven't had any users complain about it yet so I been ignoring it :shrug:).

So I do think it's not something we need to worry about here. Best effort so that (at least) our official binhost binpkgs won't go haywire with kernel checks is all I want to see really.